mypy/build.py

```diff
@@ -861,19 +865,68 @@ def is_file(path: str) -> bool:
     return res
 
 
-def find_module(id: str, lib_path_arg: Iterable[str]) -> Optional[str]:
+USER_SITE_PACKAGES = \
+    'from __future__ import print_function; import site; print(site.getusersitepackages());' \
+    'print(*site.getsitepackages(), sep="\\n")'
+VIRTUALENV_SITE_PACKAGES = \
+    'from distutils.sysconfig import get_python_lib; print(get_python_lib())'
+
+
+def call_python(python: str, command: str) -> str:
+    return subprocess.check_output([python, '-c', command], stderr=subprocess.PIPE).decode('UTF-8')
+
+
+def get_package_dirs(python: Optional[str]) -> List[str]:
+    """Find package directories for given python
+
+     This defaults to the Python running mypy."""
+    if package_dirs_cache.get(python, None):
+        return package_dirs_cache[python]
```

Comment:
Is this deliverately recomputing the cache if the contents are an empty list?

Or should this be:
```
try:
    return package_dirs_cache[python]
except KeyError:
    pass